velox/functions/lib/Re2Functions.cpp

```diff
@@ -978,7 +988,24 @@ std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern) {
       // Ensures that pattern has a single contiguous stream of wildcard
       // characters.
       if (wildcardStart != -1) {
-        return std::make_pair(PatternKind::kGeneric, 0);
+        if (singleCharacterWildcardCount) {
+          return PatternMetadata{PatternKind::kGeneric, 0};
+        }
+        // Check if the pattern is of type kFixedCenter, so the use of regular
```

Comment:
nit:"so the use of regular expressions can be avoided to match the fixed string." 
is implicit and apply to all non KGeneric